components/profile-button/logout-button.tsx

```
@@ -0,0 +1,34 @@
+function LogoutButton({ onLogout, logoutText }: LogoutButtonProps) {
+  // Memoize the click handler to prevent unnecessary re-renders
+  const handleLogout = useCallback(() => {
+    onLogout();
+  }, [onLogout]);
+
+  return (
+    <button
+      onClick={handleLogout}
+      className="group flex w-full items-center px-4 py-3 text-sm text-gray-700 dark:text-gray-300 hover:bg-gradient-to-r hover:from-red-50 hover:to-pink-50 dark:hover:from-red-900/10 dark:hover:to-pink-900/10 transition-all duration-200"
+      role="menuitem"
+    >
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)